src/main/java/org/ice4j/ice/Component.java

```diff
@@ -1079,11 +1087,20 @@ else if (keepAliveStrategy == KeepAliveStrategy.SELECTED_AND_TCP)
 
     /**
      * Sets the {@link KeepAliveStrategy} for this {@link Component}.
-     * @param strategy the strategy to set.
+     * @param strategy the strategy to set. The keep-alive strategy can only
+     * be modified before the parent {@link Agent} is started.
+     * @throws IllegalStateException if the agent has been started.
      */
     public void setKeepAliveStrategy(KeepAliveStrategy strategy)
     {
+        IceProcessingState agentState
```

Comment:
any reason to keep the setter and throw if it's too late as opposed to removing the setter and just passing the keepalive strategy in the constructor?  that would achieve the goal and prevent any chance of the caller making a mistake that would cause an exception.